test/ema.js

```diff
@@ -0,0 +1,96 @@
+/* eslint-env mocha */
+'use strict'
+
+const assert = require('assert')
+const EMA = require('../lib/ema')
+
+const emaPeriod = [3]
+const candles = [
+  { open: 6373, high: 6375.9, low: 6369.2, close: 6374.5 },
+  { open: 6374.5, high: 6408.9, low: 6366, close: 6372.9 },
+  { open: 6372.9, high: 6373.9, low: 6360.1, close: 6370.9 },
+  { open: 6370.9, high: 6400, low: 6345, close: 6400 }
+]
+
+describe('Exponential Moving Average(EMA)', () => {
+  it('sets the seed period properly', () => {
+    const ema = new EMA(emaPeriod)
+    assert.deepStrictEqual(ema._seedPeriod, 3)
+  })
+
+  it('sets the weighted multiplier properly', () => {
+    const ema = new EMA(emaPeriod)
+    assert.deepStrictEqual(ema._a, 0.5)
+  })
+
+  it('calculates ema properly when emaPeriod is 1', () => {
+    const ema = new EMA([1])
+    candles.forEach(c => ema.add(c.close))
+    assert.deepStrictEqual(ema.v(), 6400)
+    assert.deepStrictEqual(ema._buffer, [])
+    assert.deepStrictEqual(ema._values, candles.map(c => c.close))
+  })
+
+  it('updates ema value properly when emaPeriod is 1', () => {
+    const ema = new EMA([1])
+    candles.forEach(c => ema.add(c.close))
+    assert.deepStrictEqual(ema.v(), 6400)
+    ema.update(6500)
+    assert.deepStrictEqual(ema.v(), 6500)
+  })
+
+  it('calculates ema based on close price properly', () => {
+    const ema = new EMA(emaPeriod)
+    candles.forEach(c => ema.add(c.close))
+    assert.deepStrictEqual(ema.v(), 6386.15)
+  })
+
+  it('empties the buffer and ema values on reset', () => {
+    const ema = new EMA(emaPeriod)
+    candles.forEach(c => ema.add(c.close))
+    assert.deepStrictEqual(ema.v(), 6386.15)
+    assert.deepStrictEqual(ema._buffer.length, 2)
+    assert.deepStrictEqual(ema._values.length, 2)
+    ema.reset()
+    assert.deepStrictEqual(ema._buffer.length, 0)
+    assert.deepStrictEqual(ema._values.length, 0)
+  })
+
+  it('returns ema as null if the given time period is less than the provided candles', () => {
+    const ema = new EMA([candles.length + 1])
+    candles.forEach(c => ema.add(c.open))
+    assert.deepStrictEqual(ema.v(), null)
+  })
+
+  it('calculates ema properly when number of candles provided are same as the ema period', () => {
+    const ema = new EMA([candles.length])
+    candles.forEach(c => ema.add(c.high))
+    assert.deepStrictEqual(ema.v(), 6389.812)
```

Comment:
```js
var talib = require("../build/Release/talib");
var fs = require("fs");

// Display module version
console.log();
console.log("TALib Version: " + talib.version);


const marketData = [
  { open: 6373, high: 6375.9, low: 6369.2, close: 6374.5 },
  { open: 6374.5, high: 6408.9, low: 6366, close: 6372.9 },
  { open: 6372.9, high: 6373.9, low: 6360.1, close: 6370.9 },
  { open: 6370.9, high: 6400, low: 6345, close: 6400 }
].map((el) => { return el.high })

// execute EMA indicator function with time period 180
talib.execute({
    name: "EMA",
    startIdx: 0,
    endIdx: marketData.length - 1,
    inReal: marketData,
    optInTimePeriod: marketData.length
}, function (err, result) {

    // Show the result array
    console.log("EMA Function Results:");
    console.log(result);
});
```

```
TALib Version: 1.1.3
EMA Function Results:
{
  begIndex: 3,
  nbElement: 1,
  result: { outReal: [ 6389.674999999999 ] }
}
```

Reply:
values we have are:

```
6389.1
6383.02
6389.812
```

Reply:
made changes to calculate the first ema value as sma